pkg/elf/symbols.go

```diff
@@ -189,13 +210,13 @@ func (s *symbols) extractFrom(e *elf.File) error {
 	for off := uint64(0); off < strtab.Size; off += uint64(len(elfString)) {
 		// off is the offset within the string table.
 		elfString, err = stringReader.ReadString(nullTerminator)
-		if err != nil && err != io.EOF {
+		if errors.Is(err, io.EOF) {
```

Comment:
Looks like this accidentially got inverted:

```suggestion
		if !errors.Is(err, io.EOF) {
```